client/reader/following/main.tsx

```diff
@@ -16,9 +18,54 @@ import './style.scss';
 
 function FollowingStream( { ...props } ) {
 	const { currentView } = useFollowingView();
+	const { data: subscriptionsCount, isLoading: isLoadingCount } =
+		SubscriptionManager.useSubscriptionsCountQuery();
+	const { data: siteSubscriptions, isLoading: isLoadingSiteSubscriptions } =
+		SubscriptionManager.useSiteSubscriptionsQuery();
+
+	const isLoading = isLoadingCount || isLoadingSiteSubscriptions;
+
+	const hasNonSelfSubscriptions = useMemo( () => {
+		if ( isLoading ) {
+			return true;
+		}
+
+		if ( ! subscriptionsCount?.blogs || subscriptionsCount.blogs === 0 ) {
+			return false;
+		}
+
+		// If we have site subscriptions data, filter out self-owned blogs.
+		if ( siteSubscriptions?.subscriptions ) {
+			const nonSelfSubscriptions = siteSubscriptions.subscriptions.filter(
+				( sub ) => ! sub.is_owner
+			);
+			return nonSelfSubscriptions.length > 0;
+		}
+
+		return subscriptionsCount.blogs > 0;
+	}, [ subscriptionsCount?.blogs, siteSubscriptions, isLoading ] );
```

Comment:
Does using the optional chaining here impact `useMemo`? What if it resolves to `undefined`? I would suspect it would be fine and would work like we expect and re-run the function if that first item changes from `undefined` to an array or object.